packages/visitors-core/src/bottomUpTransformerVisitor.ts

```diff
@@ -22,7 +23,7 @@ export function bottomUpTransformerVisitor<TNodeKind extends NodeKind = NodeKind
     const transformerFunctions = transformers.map((transformer): BottomUpNodeTransformer => {
         if (typeof transformer === 'function') return transformer;
         return (node, stack) =>
-            getConjunctiveNodeSelectorFunction(transformer.select)(stack.getPath())
+            getConjunctiveNodeSelectorFunction(transformer.select)(stack.getPath() as NodePath<Node>)
```

Comment:
Another thing we could try here is pass `REGISTERED_NODE_KINDS` as argument to say: "get me a path whose last node may be from any node kind". That way, we are asserting the last node is not empty and we should get the appropriate return type back.

```suggestion
            getConjunctiveNodeSelectorFunction(transformer.select)(stack.getPath(REGISTERED_NODE_KINDS))
```

Reply:
Alternatively, we can use the `isFilledNodePath` helper like so.

```suggestion
            {
                const path = stack.getPath();
                if (!isFilledNodePath(path)) {
                    return false; // Or throw error as it should never happen.
                }
                return getConjunctiveNodeSelectorFunction(transformer.select)(path);
            }
```